File: Juan/P3/Java/Napakalaki/src/NapakalakiGame/Treasure.java

package NapakalakiGame;

/**
 *
 * @author juane
 */
public class Treasure {
    /**
     * Atributos
     */
    private String name;
    private int bonus;
    
    
    /* Relación */
    
    TreasureKind type; //Relación TreasureKind
    
     /** Constructor **/
    
    /**
     * Constructor que recibe como parámetros
     * @param n String     
     * @param bonus entero
     * @param t TreasureKind
     
     **/
    
    public Treasure(String n,int bonus, TreasureKind t) {
        this.name = n;
        this.bonus = bonus;
        this.type = t;
    }

   
   
    /*********** Métodos ****************/
    
    /**
     * Método getName()
     * Devuelve el nombre
     * @return name String
     */
    
    public String getName() {
        return name;
    }
    
    /**
     * Método getBonus()
     * Devuelve los bonus
     * @return bonus int
     */
     public int getBonus() {
        return bonus;
     }
     
     /**
      * Método getType()
      * Devuelve el tipo de TreasureKind
      * @return TreasureKind
      */
     public TreasureKind getType(){
         return this.type;
     }

    
     
     
    
}

File: Juan/P3/Ruby/Napakalaki/pkg/Napakalaki-0.0.1/lib/Treasure.rb
#encoding: utf-8

#Versión 3.0
module NapakalakiGame
class Treasure
    def initialize(n,bonus,t)
        @name = n
        @bonus = bonus
        @type = t
    end
   
   #MÉTODOS GET 
   attr_reader :name
   attr_reader :bonus
   attr_reader :type
   
    def to_s
        "Treasure =  #{@name}  , bonus =   #{@bonus} ,type =  #{@type}  "      
    end
end
end